driver-core/src/main/java/com/datastax/driver/core/Metadata.java

```diff
@@ -1093,13 +1128,13 @@ private Set<Host> getReplicas(String keyspace, Token token) {
         if (i >= ring.size()) i = 0;
       }
 
-      return tokenToHosts.get(ring.get(i));
+      return new ArrayList<>(tokenToHosts.get(ring.get(i)));
```

Comment:
Same remark as above. Is the new instance here needed?

Reply:
Same as https://github.com/scylladb/java-driver/pull/446#discussion_r1983607870